backend/src/main/java/com/empire/Army.java

```diff
@@ -27,22 +27,26 @@
 	String orderhint = "";
 
 	public double calcStrength(World w, Character leader, int inspires, boolean lastStand) {
-		double strength = size * (isArmy() ? 1 / 100.0 : 1);
-		double mods = 1;
+		double strength = size * (isArmy() ? Constants.armyBaseStrength : Constants.navyBaseStrength);
+
+		double mods = 1.0;
 		Region r = w.regions.get(location);
-		if (hasTag("Steel")) mods += .15;
-		if (hasTag("Seafaring") && r.isSea()) mods += 1.5;
-		if (isArmy() && !"Pirate".equals(kingdom) && w.getNation(kingdom).hasTag("Disciplined")) mods += .1;
+
+		if (hasTag(Constants.armySteelTag)) mods += Constants.steelMod;
+		if (hasTag(Constants.armySeafaringTag) && r.isSea()) mods += Constants.seafaringMod;
+		if (isArmy() && !kingdom.equals(Constants.armyPirateTag) && w.getNation(kingdom).hasTag(Constants.nationDisciplinedTag)) mods += Constants.disciplinedMod;
 		if (isArmy() && r.isLand() && NationData.isFriendly(r.kingdom, kingdom, w)) mods += r.calcFortification() - 1;
-		if (isArmy() && r.noble != null && r.noble.hasTag("Loyal") && r.kingdom.equals(kingdom)) mods += .25;
+		if (isArmy() && r.noble != Constants.noNoble && r.noble.hasTag(Constants.nobleLoyalTag) && r.kingdom.equals(kingdom)) mods += Constants.loyalMod;
```

Comment:
re: Constants.noNoble, I've found replacing null checks with named constants usually degrades readability.

Reply:
So...the rationale here was that I had noticed a region having no noble was represented with a null, a character having no captor with the empty string, etc.  I pulled them out not because I think a constant is a better solution, but to be able to mark anywhere this kind of a check happens to go back over later.  Personally I don't like the explicit null/empty string check all over the place and having to see which one to use when checking, was thinking it would be better to hide the representation internally with methods like region.hasNoble(), character.hasCaptor(), etc.  I just didn't DO this because I was trying to avoid actually doing refactoring at all (beside the simple constant replacements) and just get some tests down first (didn't entirely succeed here).

Your call though, easy enough to change them back.